packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts

```diff
@@ -912,6 +913,77 @@ export class PluginLedgerConnectorBesu
     return deployResponse;
   }
 
+  public async deployContractNoKeychain(
+    req: DeployContractSolidityBytecodeNoKeychainV1Request,
+  ): Promise<DeployContractSolidityBytecodeV1Response> {
+    const fnTag = `${this.className}#deployContract()`;
+    Checks.truthy(req, `${fnTag} req`);
+    if (isWeb3SigningCredentialNone(req.web3SigningCredential)) {
+      throw new Error(`${fnTag} Cannot deploy contract with pre-signed TX`);
+    }
+    const { contractName, contractJSONString } = req;
+    const networkId = await this.web3.eth.net.getId();
+
+    const tmpContract = new this.web3.eth.Contract(req.contractAbi);
+    const deployment = tmpContract.deploy({
+      data: req.bytecode,
+      arguments: req.constructorArgs,
+    });
+
+    const abi = deployment.encodeABI();
+    const data = abi.startsWith("0x") ? abi : `0x${abi}`;
+    this.log.debug(`Deploying "${req.contractName}" with data %o`, data);
+
+    const web3SigningCredential = req.web3SigningCredential as
+      | Web3SigningCredentialPrivateKeyHex
+      | Web3SigningCredentialCactusKeychainRef;
+
+    const runTxResponse = await this.transact({
+      transactionConfig: {
+        data,
+        from: web3SigningCredential.ethAccount,
+        gas: req.gas,
+        gasPrice: req.gasPrice,
+      },
+      consistencyStrategy: {
+        blockConfirmations: 0,
+        receiptType: ReceiptType.NodeTxPoolAck,
+        timeoutMs: req.timeoutMs || 60000,
+      },
+      web3SigningCredential,
+      privateTransactionConfig: req.privateTransactionConfig,
+    });
+
+    const { transactionReceipt: receipt } = runTxResponse;
+    const { status, contractAddress } = receipt;
+
+    if (status && contractAddress) {
```

Comment:
@jagpreetsinghsasan We could check for these being falsy and if they are, then we just throw within the if-condition. This way the nested if conditions that you have here can be flattened into just a depth of 1 instead of 2 which makes the code easier to read, debug, refactor, etc.